crates/pera-core/src/authority.rs

```diff
@@ -1522,6 +1533,28 @@ impl AuthorityState {
         Ok(())
     }
 
+    async fn handle_mpc_events(
+        &self,
+        inner_temporary_store: &InnerTemporaryStore,
+        effects: &TransactionEffects,
+        epoch_store: &Arc<AuthorityPerEpochStore>,
+    ) -> anyhow::Result<()> {
+        if !self.is_validator(epoch_store) {
+            return Ok(());
+        }
+        let status = match &effects {
+            TransactionEffects::V1(effects) => effects.status(),
+            TransactionEffects::V2(effects) => effects.status(),
+        };
+        if status.is_err() {
+            return Ok(());
+        }
```

Comment:
This feels problematic. You are returning an OK in case of an error? Should this not return `Err`?

Reply:
If the transaction failed, we don't need to handle MPC events. added comment